heuristic/repair_operators/perturbed_greedy_insert.py

```diff
@@ -0,0 +1,43 @@
+import heapq
+from numpy.random import RandomState
+
+from heuristic.classes import Route, Solution
+from heuristic.functions import create_single_customer_route
+
+
+def perturbed_greedy_insert(current: Solution,
+                            rnd_state: RandomState) -> Solution:
+    """
+    Sequentially inserts each a random permutation of the unassigned customers
+    into their y-th best, feasible route at a locally optimal leg of the tour.
+
+    Perturbed sequential best insertion in Hornstra et al. (2020).
+    """
+    rnd_state.shuffle(current.unassigned)
```

Comment:
Most of this can be merged with `greedy_insert`, using `y=1` for the usual greedy operator. As to how: look into `_worst` for the destroy operators, and how that's used.